waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/FederatedHMSHandler.java

```diff
@@ -164,18 +164,18 @@
     this.notifyingFederationService.subscribe(databaseMappingService);
   }
 
-  private ThriftHiveMetastore.Iface getPrimaryClient() throws TException {
+  private ThriftHiveMetastore.Iface getPrimaryClient() {
     return databaseMappingService.primaryDatabaseMapping().getClient();
   }
 
-  private DatabaseMapping checkWritePermissions(String databaseName) throws TException {
+  private DatabaseMapping checkWritePermissions(String databaseName) {
     DatabaseMapping mapping = databaseMappingService.databaseMapping(databaseName);
     mapping.checkWritePermissions(databaseName);
     return mapping;
   }
 
   @Override
-  public void close() throws IOException {
+  public void close() {
```

Comment:
This is overriding the method in `java.io.Closeable` which does throw an `IOException` so we should leave this.